src/extensions/default/MDNDocs/main.js

```diff
@@ -56,34 +61,39 @@ define(function (require, exports, module) {
                     try {
                         jsonData = JSON.parse(text);
                     } catch (ex) {
-                        console.error("Malformed CSS documentation database: ", ex);
+                        console.error("Malformed documentation database: ", ex);
                         result.reject();
                     }
                     result.resolve(jsonData);  // ignored if we already reject()ed above
                 })
                 .fail(function (err) {
-                    console.error("Unable to load CSS documentation database: ", err);
+                    console.error("Unable to load documentation database: ", err);
                     result.reject();
                 });
 
-            _cssDocsPromise = result.promise();
+            promiseCache[fileName] = result.promise();
         }
 
-        return _cssDocsPromise;
+        return promiseCache[fileName];
     }
 
 
     /**
-     * Inline docs provider. Currently looks up docs based on CSS properties only.
+     * Inline docs provider.
      *
      * @param {!Editor} editor
      * @param {!{line:Number, ch:Number}} pos
      * @return {?$.Promise} resolved with an InlineWidget; null if we're not going to provide anything
      */
     function inlineProvider(hostEditor, pos) {
-        var langId = hostEditor.getLanguageForSelection().getId();
-        // Only provide docs when cursor is in CSS content
-        if (langId !== "css" && langId !== "scss" && langId !== "less") {
+        var jsonFile, propInfo,
+            propQueue = [], // priority queue of propNames to try
+            langId = hostEditor.getLanguageForSelection().getId(),
+            supportedLangs = ["css", "scss", "less", "html"],
```

Comment:
Maybe this could be an `Object` or similar so we wouldn't have to rely on the index.